### PR TITLE
Expose client tools auto update for find endpoint

### PR DESCRIPTION
**Pull request by vapopov**
In this PR added exposing client auto update fields (is enabled, and required client tools version for update)

Related: https://github.com/gravitational/teleport/pull/39805/files#diff-8d3c0408dd826f27e9f13cd15f09f4d7e89202862f0b13ce131a2e3072c2a40dR49-R53

### CODE REVIEW

**Review by rosstimothy — COMMENTED** [review]
Can you add test coverage?

File: api/client/webclient/webclient.go [hunk 1]
```diff
@@ -297,6 +297,10 @@ type PingResponse struct {
 	ServerVersion string `json:"server_version"`
 	// MinClientVersion is the minimum client version required by the server.
 	MinClientVersion string `json:"min_client_version"`
+	// ToolsVersion defines the version of {tsh, tctl} for client auto update.
+	ToolsVersion string `json:"tools_version"`
+	// ToolsAutoupdate enables client auto update feature.
+	ToolsAutoupdate bool `json:"tools_autoupdate"`
```
[the remark this answers is not in the document]
Comment (vapopov) [reply api/client/webclient/webclient.go:303]: grouping fields to another struct would be better, will change this one

File: lib/web/apiserver.go [hunk 2]
```diff
@@ -1526,7 +1526,23 @@ func (h *Handler) find(w http.ResponseWriter, r *http.Request, p httprouter.Para
...
+	autoUpdateConfig, err := h.GetAccessPoint().GetAutoUpdateConfig(r.Context())
+	if err != nil && !trace.IsNotFound(err) {
+		return nil, trace.Wrap(err)
+	} else if err == nil {
+		response.ToolsAutoupdate = autoUpdateConfig.GetSpec().GetToolsAutoupdate()
+	}
+
+	autoUpdateVersion, err := h.GetAccessPoint().GetAutoUpdateVersion(r.Context())
+	if err != nil && !trace.IsNotFound(err) {
+		return nil, trace.Wrap(err)
```
[the remark this answers is not in the document]
Comment (vapopov) [reply lib/web/apiserver.go]: I added logging instead, but we might generate a lot of log messages with this error message in frequent requests to this endpoint

File: api/client/webclient/webclient.go [hunk 3]
```diff
@@ -328,6 +330,14 @@ type ProxySettings struct {
 	TLSRoutingEnabled bool `json:"tls_routing_enabled"`
 }
 
+// AutoUpdateSettings contains information about the auto update requirements.
+type AutoUpdateSettings struct {
+	// ToolsVersion defines the version of {tsh, tctl} for client auto update.
+	ToolsVersion string `json:"tools_version"`
+	// ToolsAutoupdate enables client auto update feature.
+	ToolsAutoupdate bool `json:"tools_autoupdate"`
```
[the remark this answers is not in the document]
Comment (sclevine) [reply api/client/webclient/webclient.go]: We should decide if we want to treat auto-update as a single hyphenated word (`toolsAutoupdate`/`tools_autoupdate`) or an abbreviated way of writing "automatic update" (`toolAutoUpdate`/`tools_auto_update`), and then be consistent about this.

Seems like @rosstimothy is suggesting two words, so maybe we move all of the JSON/YAML to the two-word form, and reserve `autoupdate` for the package, resource, and subcommand naming (`autoupdate_config`, `autoupdate` package, `tctl autoupdate client`?

**Review by sclevine — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**